lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/transport/netty/NettyChannelInitializer.java

```diff
@@ -24,29 +24,77 @@
 import io.netty.handler.codec.LengthFieldPrepender;
 import io.netty.handler.codec.bytes.ByteArrayDecoder;
 import io.netty.handler.codec.bytes.ByteArrayEncoder;
+import io.netty.handler.codec.http.*;
+import io.netty.handler.ssl.SslContext;
 
 /**
  * Netty channel initializer for Transport.
  */
 class NettyChannelInitializer extends ChannelInitializer<SocketChannel> {
+
+  /* Types for initiating channel */
+  public enum ChannelType {
+    NETTY, HTTP_SERVER, HTTP_CLIENT
+  }
+
   /**
    * the buffer size of the frame decoder.
    */
   public static final int MAXFRAMELENGTH = 10 * 1024 * 1024;
   private final NettyChannelHandlerFactory handlerFactory;
 
-  NettyChannelInitializer(final NettyChannelHandlerFactory handlerFactory) {
+  /**
+   * sslContext contains ssl context of the machine. used only for HTTP.
+   */
+  private final SslContext sslContext;
+
+  /**
+   * Type of channel whether it is netty or http client or http server.
+   */
+  private final ChannelType type;
+
+  NettyChannelInitializer(
+      final NettyChannelHandlerFactory handlerFactory,
+      final SslContext sslContext,
```

Comment:
If this can be null, how about creating another constructor not receiving this variable?